### PR TITLE
 Customer related values are NULL for guests converted to customers after checkout. #19166

### PR DESCRIPTION
**Pull request by Nazar65**
### Description (*)
Customer name and all customer related values should not be NULL

### Fixed Issues (if relevant)
1. magento/magento2#19166:Customer related values are NULL for guests converted to customers after checkout.
### Manual testing scenarios (*)

1. Allow customers to create account after checkout
2. Create few orders as guest customers and create an account after checkout process
3. Create a module with a plugin aroundSave for SalesOrder class
4. In aroundSave method, try to fetch customer name from order object
```
public function aroundSave(
        OrderResource $subject,
        \Closure $proceed,
        SalesOrder $order
    ) {
        [...]
        if ($order->getCustomerId()) {
            $customerName = $order->getCustomerName(); // returns Customer Name
        [...]
        }
    }
```

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @Nazar65. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Quote/Model/QuoteManagement.php [hunk 1]
```diff
@@ -354,6 +354,12 @@ public function placeOrder($cartId, PaymentInterface $paymentMethod = null)
         if ($quote->getCheckoutMethod() === self::METHOD_GUEST) {
             $quote->setCustomerId(null);
             $quote->setCustomerEmail($quote->getBillingAddress()->getEmail());
+            $quote->setCustomerFirstname($quote->getBillingAddress()->getFirstname());
+            $quote->setCustomerLastname($quote->getBillingAddress()->getLastname());
+
+            if ($quote->getBillingAddress()->getMiddlename()) {
+                $quote->setCustomerMiddlename($quote->getBillingAddress()->getMiddlename());
```
Comment (orlangur) [inline app/code/Magento/Quote/Model/QuoteManagement.php]: Maybe set them just when they are equal to `null`?

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Quote/Model/QuoteManagement.php (hunk 1 above)
Comment (Nazar65) [reply app/code/Magento/Quote/Model/QuoteManagement.php]: hmm, yes it would be better. ok

Comment (Nazar65) [reply app/code/Magento/Quote/Model/QuoteManagement.php]: All done. also fixed the failed code style test

**Comment by Nazar65** [comment]
@orlangur I'm have made changes.

**Comment by slavvka** [comment]
This PR is a duplicate of #18930

**Comment by Nazar65** [comment]
@slavvka But i'm have a better solution, and i'm already cover all changes with the integration tests.

**Comment by Nazar65** [comment]
Hi @orlangur any updates ?

**Review by orlangur — APPROVED** [review]
@Nazar65 didn't plan to process this one, but anyway... Looks good to me!

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3477](https://jira.corp.magento.com/browse/ENGCOM-3477) has been created to process this Pull Request

**Pull request merged**